### PR TITLE
Increase Test Code Coverage of Backend

### PR DESCRIPTION
**Pull request by diogo-botelho**
Added tests for following files:

- routes/eventRoutes
Some tests fail because we need to implement json schema.

- routes/auth:
Some tests fail because we need to implement json schema.

- App

- Config

- middleware/auth:
Need to revisit Typescript best practices. Issue opened: #32

### CODE REVIEW

File: backend/src/config.test.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    expect(config.DB_URI).toEqual("other");
+    expect(config.BCRYPT_WORK_FACTOR).toEqual(12);
+
+    delete process.env.SECRET_KEY;
+    delete process.env.PORT;
+    delete process.env.BCRYPT_WORK_FACTOR;
+    delete process.env.DATABASE_URL;
+
+    console.log(config.DB_URI);
+    expect(config.DB_URI).toEqual("jobly");
```
Comment (samau3) [inline backend/src/config.test.ts]: Rename `jobly`

File: backend/src/config.test.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    
+    console.log(config.DB_URI);
+    process.env.NODE_ENV = "test";
+    
+    console.log(config.DB_URI);
+    expect(config.DB_URI).toEqual("jobly_test");
```
Comment (samau3) [inline backend/src/config.test.ts]: Rename `jobly`

File: backend/src/routes/authRoutes.test.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+//   test("bad request with invalid data", async function () {
+//     const resp = await request(app)
+//         .post("/auth/login")
+//         .send({
+//           username: 42,
+//           password: "above-is-a-number",
+//         });
+//     expect(resp.statusCode).toEqual(400);
+//   });
+});
```
Comment (samau3) [inline backend/src/routes/authRoutes.test.ts:79]: Might be better to leave these tests as failing to remind us to add `jsonschema`?

File: backend/src/routes/eventRoutes.test.ts [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+  //WHEN WE INTRODUCE BUSKERS MODEL
+//   test("unauth for others", async function () {
+//     const resp = await request(app)
+//       .patch(`/events/${testEventIds[0]}`)
+//       .send({
+//         title: "New title",
+//       })
+//       .set("authorization", `Bearer ${u1Token}`);
+//     expect(resp.statusCode).toEqual(401);
+//   });
```
Comment (samau3) [inline backend/src/routes/eventRoutes.test.ts]: Similar idea to leave the failing tests for `jsonschema`, might be better to allow this test to fail to remind us to implement this feature

File: backend/src/config.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
 export const PORT = +process.env.PORT || 3001;
 
-export const DB_URI =
-  process.env.NODE_ENV === "test"
-    ? "postgresql://diogobotelho:password@localhost/busk_app_test"
-    : process.env.DATABASE_URL ||
-      "postgresql://diogobotelho:password@localhost/busk_app";
-//   "busk_app_test"
-// : "busk_app";
+export function getDatabaseUri() {
```
Comment (samau3) [inline backend/src/config.ts:3]: Resolves the failing of the config tests - to discuss if needed.

File: backend/src/middleware/auth.test.ts [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  test("works: via header", function () {
+    expect.assertions(2);
+    const req = {
+      headers: { authorization: `Bearer ${testJwt}` },
+    };
+    const res = { locals: {} };
+    const next = function (err: ExpressError) {
+      expect(err).toBeFalsy();
+    };
+    authenticateJWT(req as Request, res as Response, next as NextFunction);
```
Comment (samau3) [inline backend/src/middleware/auth.test.ts:26]: Issue was that `req`, `res`, and `next` were not properly being treated as express `Request`, etc. The solution is to type cast them as such so Typescript knows that they should be treated as such. 

https://javascript.plainenglish.io/how-to-unit-test-express-middleware-typescript-jest-c6a7ad166e74

File: backend/src/middleware/auth.test.ts [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    expect(res.locals).toEqual({});
+  });
+});
+
+describe("ensureLoggedIn", function () {
+  test("works", function () {
+    expect.assertions(1);
+    const req = {};
+    const res: unknown = { locals: { user: { username: "test" } } };
```
Comment (samau3) [inline backend/src/middleware/auth.test.ts:63]: Utilizing `unknown` for various `req` and `res` when type casting as `Request` and `Response` weren't enough. Need to look this up to see if it's best practice. 

TS Error

File: backend/src/config.ts (hunk 5 above)
Comment (diogo-botelho) [reply backend/src/config.ts:3]: Cool

**Pull request merged**